static/src/javascripts/projects/common/modules/article/space-filler.js

```diff
@@ -1,66 +1,47 @@
-/* jscs:disable disallowDanglingUnderscores */
 define([
-    'common/utils/$',
-    'common/utils/fastdom-idle',
+    'common/utils/fastdom-promise',
     'Promise',
     'raven',
     'common/modules/article/spacefinder'
 ], function (
-    $,
-    idleFastdom,
+    fastdom,
     Promise,
     raven,
     spacefinder
 ) {
-    return new SpaceFiller();
+    var lastIteration = Promise.resolve(false);
```

Comment:
In an ideal worldI suppose I would rather declare my modules as constructors, and then instantiate them in some kind of service locator. I definitely agree that it is not a good idea to change the API for the sake of testing.

That's why I originally implemented space-filler as an instance of a constructor - so I could avoid a test interface whilst allowing tests to create their own module instances with the `object.constructor` approach. I don't know if it's the ideal approach, or if it solves every problem. The most obvious alternative, I guess, is to write the test utils as pseudo-privates (`_reset`, etc.), but then you have to wrap everything in eslint switches, which feels a bit awkward.

How were modules written in your last project?